### ISSUE TITLE
operator-sdk build eunomia-operator command does not work

### ISSUE THREAD

**Issue by sbar95** [KohlsTechnology/eunomia#174]
**What version of eunomia are you using?**

eunomia version: v0.0.4

**Does this issue reproduce with the latest release?**

yes

**What operating system and processor architecture are you using (`kubectl version`)?**

n/a

**What did you do?**

I issue `operator-sdk build eunomia-operator` command in the root of project directory; steps as in [DEVELOPMENT.md](https://github.com/KohlsTechnology/eunomia/blob/master/DEVELOPMENT.md#building-the-image-on-your-local-workstation).

**What did you expect to see?**

```
INFO[0000] Building OCI image eunomia-operator          
Sending build context to Docker daemon  72.34MB
Step 1/4 : FROM registry.access.redhat.com/ubi8/ubi:latest
 ---> cb642e6a9917
...
Step 4/4 : ENTRYPOINT ["/usr/local/bin/eunomia-operator"]
 ---> Using cache
 ---> f8ecdd0f1171
Successfully built f8ecdd0f1171
Successfully tagged eunomia-operator:latest
INFO[0003] Operator build complete.
```

**What did you see instead?**

```
INFO[0000] Building OCI image eunomia-operator          
Sending build context to Docker daemon  72.34MB
Step 1/4 : FROM registry.access.redhat.com/ubi8/ubi:latest
 ---> cb642e6a9917
Step 2/4 : ENV OPERATOR=/usr/local/bin/eunomia-operator
 ---> Using cache
 ---> 69e521bf837b
Step 3/4 : COPY _output/bin/eunomia ${OPERATOR}
COPY failed: stat /var/lib/docker/tmp/docker-builder954872600/_output/bin/eunomia: no such file or directory
Error: failed to output build image eunomia-operator: (failed to exec []string{"docker", "build", "-f", "build/Dockerfile", "-t", "eunomia-operator", "."}: exit status 1)
Usage:
  operator-sdk build <image> [flags]

Flags:
  -h, --help                      help for build
      --image-build-args string   Extra image build arguments as one string such as "--build-arg https_proxy=$https_proxy"
      --image-builder string      Tool to build OCI images. One of: [docker, buildah] (default "docker")

Global Flags:
      --verbose   Enable verbose logging
```

**Reason and proposed solution**

Build context in [Docker usage documentation](https://docs.docker.com/engine/reference/builder/#usage):
> Traditionally, the Dockerfile is called Dockerfile and located in the root of the context. You use the `-f` flag with `docker build` to point to a Dockerfile anywhere in your file system.

[Docker build manual](https://docs.docker.com/engine/reference/commandline/build/#options):
> `--file , -f` Name of the Dockerfile (Default is ‘PATH/Dockerfile’)

- In eunomia v0.0.4 [such build context](https://github.com/KohlsTechnology/eunomia/blob/v0.0.4/scripts/build-images.sh#L44) and [such COPY path in Dockerfile](https://github.com/KohlsTechnology/eunomia/blob/v0.0.4/build/Dockerfile#L20) are used.
- In eunomia v0.0.1 [this build context](https://github.com/KohlsTechnology/eunomia/blob/v0.0.1/scripts/build-images.sh#L25) and [this COPY path in Dockerfile](https://github.com/KohlsTechnology/eunomia/blob/v0.0.1/build/Dockerfile#L20) were used.

- In v0.0.1 build context was in the root of the eunomia project and Dockerfile was pointed to explicitly with the `-f` flag, which has many advantages (see [this stackoverflow answer](https://stackoverflow.com/a/34392052)).
- In v0.0.4 build context is `eunomia/build` directory, and no `-f` flag is passed (so the path to Dockerfile is set implicitly to `eunomia/build/Dockerfile`.

`operator-sdk build` command has been always using the first approach. Putting all advantages and disadvantages aside, eunomia-operator is using **operator-framework**, which operator-sdk is part of. So I think `operator-sdk build` command should work.

In order to achieve this, we should remake [Dockerfile](https://github.com/KohlsTechnology/eunomia/blob/master/build/Dockerfile) and [build images script](https://github.com/KohlsTechnology/eunomia/blob/master/scripts/build-images.sh) to adopt the first approach again to be **operator-sdk**-compliant.

**Comment by sbar95** [comment]
@seanmalloy This bug is blocking the story #162 because of the approach I've taken there - I wanted to include default job-template files in eunomia image with a Dockerfile COPY command without copying them to `eunomia/build` directory. I can do it the way @raffaelespazzoli suggested [here](https://github.com/KohlsTechnology/eunomia/issues/162#issuecomment-558724239) but I think my approach requires less code and doesn't duplicate the files into go code.

**Comment by akavel** [comment]
Honest question: does `operator-sdk build` bring us any value over what `go build` does?

Unless it does, I prefer the simplicity of `go build`; running it through `operator-sdk build` brings one more layer of indirection and complexity. Thus, personally, I'd actually prefer if we resolved this issue by removing the mention of `operator-sdk build` from the repository... and by then on gleefully pretending it was maybe even never there at all? ;D

**Comment by sbar95** [comment]
I think it doesn't bring anything new above what `go build` does.

I agree with the Dockerfile being in the root of context as a more comprehensible solution. I, for a log time, have been using docker 1.13.1 where there was no such feature as separating context from Dockerfile. Only after a quick research I found out it's now possible.

I just want to be operator-sdk compliant as I wrote in the description. If we don't have to be compliant, I have no problem with it as I also rather use `make`. Just wanted to make `operator-sdk` work again ;)
